### PR TITLE
adds jwe-encryption in scoped_keys

### PR DESCRIPTION
**Pull request by tarikeshaq**
### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - `automation/all_tests.sh` runs to completion and produces no failures
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/master/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

Separated from #3187 to make review easier. **Currently generates compiler warnings** since encrypt_keys_jwe is not consumed anywhere, added an #[allow(dead_code)] that will be removed in #3187.

### CODE REVIEW

File: components/fxa-client/src/scoped_keys.rs [hunk 1]
```diff
@@ -88,6 +103,66 @@ impl ScopedKeysFlow {
...
+            .ok_or_else(|| ErrorKind::ParseJsonError("Could not find x in jwk json"))?
+            .as_str()
+            .ok_or_else(|| ErrorKind::ParseJsonError("x value is not a string in jwk json"))?;
+        let y = jwk_value
+            .as_object()
+            .ok_or_else(|| ErrorKind::ParseJsonError("jwk json is not an object"))?
+            .get("y")
+            .ok_or_else(|| ErrorKind::ParseJsonError("Could not find y in jwk json"))?
+            .as_str()
+            .ok_or_else(|| ErrorKind::ParseJsonError("y value is not a string in jwk json"))?;
```
Comment (eoger) [inline components/fxa-client/src/scoped_keys.rs:121]: Can we replace all of this with `let jwk: Jwk = serde_json::from_str(jwk)`?

Comment (tarikeshaq) [reply components/fxa-client/src/scoped_keys.rs:121]: Yes we can 🤦 I added the Jwk struct after I had written this stuff. Thanks @eoger

**Comment by rfk** [comment]
Adding @linuxwolf for visibility here given his deep familiarity with all things JOSE.

**Comment by tarikeshaq** [comment]
Thanks a ton, @linuxwolf!! I Appreciate the comments and feedback! 😄

**Pull request merged**